csharp/ql/src/semmle/code/csharp/dispatch/Dispatch.qll

```diff
@@ -387,7 +387,8 @@ private module Internal {
       result = this.getAStaticTarget()
       or
       result.getUnboundDeclaration() =
-        this.getASubsumedStaticTarget0(Gvn::getGlobalValueNumber(result.getDeclaringType()))
+        this.getASubsumedStaticTarget0(pragma[only_bind_out](Gvn::getGlobalValueNumber(result
+                .getDeclaringType())))
```

Comment:
How about instead changing
```ql
cached
GvnType getGlobalValueNumber(Type t) { ... }
```

to
```ql
cached
GvnType getGlobalValueNumberImpl(Type t) { ... }

pragma[inline]
GvnType getGlobalValueNumber(Type t) {
  result = pragma[only_bind_out](getGlobalValueNumberImpl(t))
}
```

in `Unification.qll`?

Reply:
That seems to break other code. Specifically `typeConstraintUnifiable/typeConstraintSubsumes` in `Unification.qll` where we rely on doing `getGlobalValueNumber` backwards and forcing the optimiser not to do that makes the code awful. It looks we avoid blowup in those cases due to how small `TTypeConstraint` is.

Reply:
OK, let's get this in now, and then I can investigate the other approach in a follow-up PR.